### PR TITLE
Add randomized context store

### LINKED ISSUE #506: Create `ExecutionPolicy` to randomly select bugs
If we want to have a very simple "distributed" system, we can run a bunch of caffeine instances independently on different computers. In order to prevent repetition of work, we can randomly select Contexts. This will still be wildly inefficient, but it's a step in the right direction.

### PR DESCRIPTION
**Pull request by Skrinikov**
WIP just want to see if it builds
Fixes #506

### CODE REVIEW

File: local.bazelrc [hunk 1]
```diff
@@ -0,0 +1 @@
+build '--action_env=PATH=/home/danieil/.vscode-server/bin/b3318bc0524af3d74034b8bb8a64df0ccf35549a/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Program Files/AdoptOpenJDK/jdk-11.0.5.10-hotspot/bin:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files/Git/cmd:/mnt/c/mingw-w64/mingw32/bin:/mnt/c/Strawberry/c/bin:/mnt/c/Strawberry/perl/site/bin:/mnt/c/Strawberry/perl/bin:/mnt/c/Users/skrin/AppData/Local/Programs/Python/Python37-32/Scripts/:/mnt/c/Users/skrin/AppData/Local/Programs/Python/Python37-32/:/mnt/c/Users/skrin/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/skrin/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/texlive/2019/bin/win32:/mnt/c/mingw-w64/mingw32/bin:/mnt/c/dev_tools/kotlinc/bin:/mnt/c/Program Files/Java/jdk-12.0.2/bin:/mnt/c/dev_tools/IntelliJ IDEA 2019.3.1/bin:/mnt/c/Users/skrin/AppData/Local/Programs/MiKTeX 2.9/miktex/bin/x64/:/mnt/c/Users/skrin/AppData/Local/Microsoft/WindowsApps'
```
Comment (Phantomical) [inline local.bazelrc:1]: Can you remove this file from the PR. It's in `.gitignore` on master but I don't think you've rebased in a while.

File: include/caffeine/Interpreter/Store.h [hunk 2]
```diff
@@ -75,4 +77,29 @@ class QueueingContextStore : public ExecutionContextStore {
   std::queue<Context> queue;
 };
 
+class RandomizedContextStore : public ExecutionContextStore {
```
Comment (Phantomical) [inline include/caffeine/Interpreter/Store.h:80]: Nit: Could you add a small doc comment outlining what this does. The name is pretty clear but I'm trying to move towards having doc comments for more things.

**Review by Phantomical — APPROVED** [review]
LGTM with two small nits.

**Comment by Phantomical** [comment]
@Skrinikov Are you going to merge this at some point?

**Comment by mishazharov** [comment]
@Skrinikov what's the status of this PR?

**Pull request merged**